Testreglar/4.1.2/Nett/nett-4.1.2a.json

```diff
@@ -0,0 +1,451 @@
+			"type": "jaNei",
+			"ruting": {
+				"nei": {
+					"type": "avslutt",
+					"fasit": "Nei",
+					"utfall": "Skjemaelementetet har tilgjengelig navn og er kodet med riktig rolle. Skjemaelementet er plassert i en gruppe som ikke er identifisert i koden."
+				},
+				"ja": {
+					"type": "gaaTil",
+					"steg": "3.14"
+				}
+			}
+		},
+		{
+			"stegnr": "3.14",
+			"spm": "Har elementet med rolle group et tilgjengelig navn, som ikke er tomt?",
+			"ht": "<p>Slik tester du:</p>\n<ul>\n<li>Åpne Accessibility Checker i DevTools i nettleseren (F12) Chrome</li>\n<li>Sjekk at informasjon under \"Name\" ikke er tomt.</li>\n</ul>",
+			"type": "jaNei",
+			"ruting": {
+				"ja": {
+					"type": "gaaTil",
+					"steg": "3.15"
+				},
+				"nei": {
+					"type": "avslutt",
+					"fasit": "Nei",
+					"utfall": "Skjemaelementet har tilgjengelig navn og er kodet med riktig rolle. Skjemaelement er plassert i en gruppe som ikke har et tilgjengelig navn."
+				}
+			}
+		},
+		{
+			"stegnr": "3.15",
+			"spm": "Blir skjemaelementet identifisert av det tilgjengelige navnet for gruppen kombinert med skjemaelementets tilgjengelige navn?",
+			"ht": "<p>Du skal vurdere både skjemaelementets tilgjengelig navn og det tilgjengelige navnet for gruppen som skjemaelementet ligger inne i samlet. </p>",
+			"type": "jaNei",
+			"ruting": {
+				"ja": {
+					"type": "gaaTil",
+					"steg": "3.16"
+				},
+				"nei": {
+					"type": "avslutt",
+					"fasit": "Nei",
+					"utfall": "Skjemaelementetet har tilgjengelig navn og er kodet med riktig rolle. Skjemaelementet er plassert i en gruppe som har et tilgjengelig navn, men det tilgjengelige navnet for gruppen identifiserer ikke formålet med skjemaelementet."
+				}
+			}
+		},
+		{
+			"stegnr": "3.16",
+			"spm": "Har skjemaelementet mer enn en tilstand når brukeren samhandler med den?",
+			"ht": "<p>En tilstand er en dynamisk egenskap som endres når brukeren samhandler med skjemaelementet.</p>\n<p>Trykk på skjemaelementet og sjekk om elementet har minst en av disse tilstandene: </p>\n<ul>\n<li>avkryssingsboks eller radioknapp: checkbox eller radio - avkrysset eller ikke avkrysset</li>\n<li>nedtrekksliste: combobox - utvidet eller sammensluttet</li>\n<li>inndatafelt: textbox - utfylt eller ikke utfylt</li>\n<li>switch: switch - av/på</li>\n<li>annet: andre relevante tilstander (State) knyttet til <a href=\"https://www.w3.org/TR/wai-aria-1.2/#widget_roles\" target=\"_blank\" rel=\"noopener\">Widget Roles</a></li>\n</ul>",
+			"type": "jaNei",
+			"ruting": {
+				"nei": {
+					"type": "avslutt",
+					"fasit": "Ja",
+					"utfall": "Skjemaelementet har et tilgjengelig navn, og riktig rolle,  som beskriver formålet med det aktuelle elementet."
+				},
+				"ja": {
+					"type": "gaaTil",
+					"steg": "3.17"
+				}
+			}
+		},
+		{
+			"stegnr": "3.17",
+			"spm": "Hvilken tilstand tester du?",
```

Comment:
Vil dette fungere med flere tilstander?

Reply:
Testregelen fungerer for flere tilstander ref. steg 3.16. Hvis et element har kun en tilstand så avsluttes testen i steg 3.16.

Reply:
Steg 3.17
- Hvor mange tilstander er det? Antall tilstander
- Hvor mange tilstander er ikke programmatisk bestemt? hvis svaret er null, så er dette samsvar.
- Beskrive tilstander som ikke er programmatisk bestemt.